### ISSUE TITLE
Timeout after 15 seconds in CONFIGURATION state (1.21 backend server)

### ISSUE THREAD

**Issue by scrayos** [PaperMC/Velocity#1367]
### Expected Behavior

I expect clients to be able to stay in configuration phase indefinitely. There are [special keep alive packets](https://wiki.vg/Protocol#Clientbound_Keep_Alive_.28configuration.29) that should allow clients to stay connected that way, without any limitation. 

### Actual Behavior

Clients are always disconnected after 15 seconds.

### Steps to Reproduce

1. Add the listener below
2. Connect to velocity
3. Get disconnected after 15 seconds

```java
    @Subscribe
    public EventTask onConfigurationWait(@NotNull final PlayerFinishConfigurationEvent e) {
        return EventTask.resumeWhenComplete(new CompletableFuture<>());
    }
 ```

### Plugin List

Only the test plugin

### Velocity Version

[22:58:43 INFO]: Velocity 3.3.0-SNAPSHOT (git-aa4e8780-b401)
[22:58:43 INFO]: Copyright 2018-2023 Velocity Contributors. Velocity ist lizenziert unter den Bedingungen der GNU General Public License v3.
[22:58:43 INFO]: velocitypowered.com - GitHub

**Comment by Gerrygames** [comment]
I can't reproduce this during the initial configuration. This seems to only affect reconfiguration when switching servers.

**Comment by scrayos** [comment]
> I can't reproduce this during the initial configuration. This seems to only affect reconfiguration when switching servers.

I actually only tested this for the initial connection. I installed this listener and then joined the server. After 15 seconds, I reliably time out. I did not even test reconfiguration on server-switch, so that's weird. And you were able to stay connected this way indefinitely on initial connect?

Did you modify the configuration on either velocity or your backend server? (I kept all defaults)

**Comment by Gerrygames** [comment]
This was an issue when the configuration API was still WIP. Users that took longer than 30 seconds to download our resource pack timed out. I did fix this before the configuration API was merged. I explicitly tested this again by holding the PlayerFinishConfigurationEvent for 30 seconds and could not reproduce this issue. I am running Paper 1.20.4 and tested with 1.20.4 and 1.21 client (with ViaVersion). There aren't any modifications done to the configuration progress. I will test again with Paper 1.21, maybe there is some kind of difference.

**Comment by Gerrygames** [comment]
This issue does indeed occur with Paper 1.21 as the backend server. Apparently Mojang changed how the switch from configuration to play state works and does not send any keep alive packets in this limbo state anymore.

**Comment by scrayos** [comment]
> This issue does indeed occur with Paper 1.21 as the backend server. Apparently Mojang changed how the switch from configuration to play state works and does not send any keep alive packets in this limbo state anymore.

Yeah, I also used 1.21 as the backend. Gonna change the title of the issue.

**Comment by scrayos** [comment]
According to [the wiki](https://wiki.vg/Protocol#Clientbound_Keep_Alive_.28configuration.29) the 15 second timeout indicates, that no keep alive packet was received from the client. The client would disconnect after 20 seconds, if it didn't receive any keep-alives. So maybe, the keep-alive packets from the client are not correctly forwarded to the backend? :thinking:

**Comment by Gerrygames** [comment]
Yeah so I think this is unfortunately a won't fix. The way vanilla server has implemented the protocol switching since 1.20.5 makes it impossible for the backend server to send any packets while switching protocols. Thus it won't send any keep alive to which the client could reply and will time out. I will add a note that this event should not be held for more than 10-15 seconds. As a replacement I will make the PlayerEnteredConfigurationEvent an awaiting event. There shouldn't be any issue with holding the client and server in that state.

**Comment by Gerrygames** [comment]
This can now be fixed by using the new PlayerConfigurationEvent.
